infra/core/compute/ubuntu-jumphost.bicep

```diff
@@ -0,0 +1,277 @@
+targetScope = 'resourceGroup'
+
+/*
+** Ubuntu VM Jumphost
+** Copyright (C) 2023 Microsoft, Inc.
+** All Rights Reserved
+**
+***************************************************************************
+**
+** Creates an Ubuntu VM with appropriate capabilities to act as a
+** jumphost
+*/
+
+// ========================================================================
+// USER-DEFINED TYPES
+// ========================================================================
+
+// From: infra/types/DiagnosticSettings.bicep
+@description('The diagnostic settings for a resource')
+type DiagnosticSettings = {
+  @description('The number of days to retain log data.')
+  logRetentionInDays: int
+
+  @description('The number of days to retain metric data.')
+  metricRetentionInDays: int
+
+  @description('If true, enable diagnostic logging.')
+  enableLogs: bool
+
+  @description('If true, enable metrics logging.')
+  enableMetrics: bool
+}
+
+// ========================================================================
+// PARAMETERS
+// ========================================================================
+
+@description('The diagnostic settings to use for logging and metrics.')
+param diagnosticSettings DiagnosticSettings
+
+@description('The name of the primary resource')
+param name string
+
+@minLength(3)
+@maxLength(15)
+@description('The name of the linux PC.  By default, this will be automatically constructed by the resource name.')
+param computerLinuxName string?
+
+@description('Username for the Virtual Machine.')
+param adminUsername string
+
+@description('Type of authentication to use on the Virtual Machine. SSH key is recommended.')
+@allowed([
+  'sshPublicKey'
+  'password'
+])
+param authenticationType string = 'password'
+
+@description('SSH Key or password for the Virtual Machine. SSH key is recommended.')
+@secure()
+param adminPasswordOrKey string
+
+@description('The Ubuntu version for the VM. This will pick a fully patched image of this given Ubuntu version.')
+@allowed([
+  'Ubuntu-1804'
+  'Ubuntu-2004'
+  'Ubuntu-2204'
+])
+param ubuntuOSVersion string = 'Ubuntu-2204'
```

Comment:
Yes, we can pick one for our scenario but left this as-is since it's one of the core modules and we want to encourage composability. Also, the images listed in this set of options are limited to those that support the `Trusted Launch` feature for security reasons. And as a result, this impacts the way other features are installed (such as dotnet) due to their availability in the package feed.